File: week10/20221028/숫자-문자열과-영단어/java/src/Solution.java
import java.util.Arrays;

public class Solution {
    public final String[] numbers = {
            "zero", "one", "two", "three", "four",
            "five", "six", "seven", "eight", "nine"
    };

    public String[] index = {
            "0", "1", "2", "3", "4", "5", "6", "7", "8", "9"
    };

    public int solution(String s) {
        String answer = Arrays.stream(index)
                .reduce(s, (acc, index) -> acc.replace(numbers[Integer.parseInt(index)], index));

        return Integer.parseInt(answer);
    }
}

File: week10/20221028/숫자-문자열과-영단어/javascript/solution.js
function solution(s) {
  const numbers = [
    "zero", "one", "two", "three", "four",
    "five", "six", "seven", "eight", "nine"
  ]

  var answer = numbers.reduce((acc, cur, index) => acc.replaceAll(cur, index), s)

  return Number(answer);
}